### PR TITLE
#0: Fix multi-cq for FD2.0, enable tests

### PR DESCRIPTION
**Pull request by tt-dma**
Two bugfixes + some additional test fixes, multi-cq should be up and running now
Passing CI: https://github.com/tenstorrent-metal/tt-metal/actions/runs/8624837068

### CODE REVIEW

File: tt_metal/impl/dispatch/kernels/cq_dispatch.cpp [hunk 1]
```diff
@@ -106,7 +107,8 @@ void completion_queue_reserve_back(uint32_t num_pages) {
 
 FORCE_INLINE
 void notify_host_of_completion_queue_write_pointer() {
-    uint64_t pcie_address = get_noc_addr_helper(pcie_noc_xy_encoding, HOST_CQ_COMPLETION_WRITE_PTR);  // For now, we are writing to host hugepages at offset
+    uint64_t completion_queue_write_ptr_addr = command_queue_base_addr + HOST_CQ_COMPLETION_WRITE_PTR;
```
Comment (abhullar-tt) [inline tt_metal/impl/dispatch/kernels/cq_dispatch.cpp:110]: is this the same as `completion_queue_base_addr`? if thats the case then we wouldn't need that extra compile time arg

Comment (tt-dma) [reply tt_metal/impl/dispatch/kernels/cq_dispatch.cpp:110]: Maybe it's named bad, this is the write ptr in the reserved region, not actually in the completion q region

Comment (abhullar-tt) [reply tt_metal/impl/dispatch/kernels/cq_dispatch.cpp:110]: oh this is the address where we write the completion queue addr?

Comment (tt-dma) [reply tt_metal/impl/dispatch/kernels/cq_dispatch.cpp:110]: Yes that's right, a bug here is that it was being hard-coded - need to pass it in for multi cq

**Pull request merged**